### ISSUE TITLE
Electron Mac OSX crash render when write file too fast

### ISSUE THREAD

**Issue by 25512345** [atom/electron#3814]
```
var fs = require('fs');
    var path = require('path');
    for (var i = 0; i < 1000; i++){
      fs.open(path.join(__dirname, "test.txt"), "a", function(err, fd) {
       if (err) {
         return;
       }
       fs.writeFile(fd, i + "\n");
     });
    }
```
When run code above, DOM tree gone, DevTool show dialog: "Detached from the target. Remote debugging has been terminated with reason: Render process gone. Please re-attach to the new target"

**Comment by zcbenz** [comment]
This is a duplicate of #3786.

**Comment by 25512345** [comment]
Does this issues fix on electron 0.36.1? I try this code on electron 0.36.1 and render process still crash

**Comment by zcbenz** [comment]
I can not reproduce with your code snippet, I pasted it into devtools console but it works fine.

Can you paste the crash report that can be found in `Console.app`?
